yang/frr-bgp-common.yang

```diff
@@ -0,0 +1,1094 @@
+    leaf as-path-filter-list-export {
+      type frr-bt:as-path-filter-ref;
+    }
+  }
+
+  grouping route-selection-options {
+    description
+      "Configuration relating to route selection options.";
+    container route-selection-options {
+      description
+        "Parameters relating to options for route selection.";
+      leaf always-compare-med {
+        type boolean;
+        default "false";
+        description
+          "Compare multi-exit discriminator (MED) value from
+           different ASes when selecting the best route.  The
+           default behaviour is to only compare MEDs for paths
+           received from the same AS.";
+      }
+
+      leaf deterministic-med {
+        type boolean;
+        default "false";
+        description
+          "Compare multi-exit discriminator (MED) value from
+           Same ASes when selecting the best route.";
+      }
+
+      leaf confed-med {
+        type boolean;
+        default "false";
+        description
+          "Compare multi-exit discriminator (MED) value from
+           different Sub ASes when selecting the best route.";
+      }
+
+      leaf missing-as-worst-med {
+        type boolean;
+        default "false";
+        description
+          "Missing multi-exit discriminator (MED) value is
+           considered as value of infinity, making the path
+           least desirable when selecting the best route.";
+      }
+
+      leaf aspath-confed {
+        type boolean;
+        default "false";
+        description
+          "Compare path lengths including confederation sets
+           and sequences in selecting a route.";
+      }
+
+      leaf ignore-as-path-length {
+        type boolean;
+        default "false";
+        description
+          "Ignore the AS path length when selecting the best path.
+           The default is to use the AS path length and prefer paths
+           with shorter length.";
+      }
+
+      leaf external-compare-router-id {
+        type boolean;
+        default "true";
+        description
+          "When comparing similar routes received from external BGP
+           peers, use the router-id as a criterion to select the
+           active path.";
+      }
+
+      leaf allow-multiple-as {
+        type boolean;
+        default "false";
+        description
+          "Allow multi-path to use paths from different neighbouring
+           ASes.  The default is to only consider multiple paths
+           from the same neighbouring AS.";
+      }
+
+      leaf multi-path-as-set {
+        when "../allow-multiple-as = 'true'";
+        type boolean;
+        default "false";
+        description
+          "Multi-path with AS-SET, When set to 'true' it adds AS set
+           information for aggregate routes, When set to 'false' it
+           prevents AS set generation.";
+      }
+    }
+  }
+
+  grouping med-config {
+    description
+      "Configuration relating to MED.";
+    container med-config {
```

Comment:
This container should have presence statement.

Reply:
As discussed, I will check other places too.